post-processor/README.md

```diff
@@ -0,0 +1,135 @@
+
+Historically, Packer has used a JSON template for its configuration. From version 1.7.0, HCL2 becomes officially the preferred template configuration format. You can find examples on how to use HCL Templates at `/examples` folder.
+
+This is a basic Packer Template used to create a custom CentOS image on IBM Cloud - VPC
+
+```hcl
+packer {
+  required_plugins {
+    ibmcloud = {
+      version = ">=v3.0.0"
+      source = "github.com/IBM/ibmcloud"
+    }
+  }
+}
+
+variable "ibm_api_key" {
+  type = string
+  default = "${env("IBM_API_KEY")}"
+}
+
+locals {
+  timestamp = regex_replace(timestamp(), "[- TZ:]", "")
+}
+
+source "ibmcloud-vpc" "centos" {
+  api_key = "${var.ibm_api_key}"
+  region = "au-syd"
+
+  subnet_id = "02h7-9645d633-55a8-463c-b3b3-5cd302f2ee32"
+  resource_group_id = ""
+  security_group_id = ""
+
+  vsi_base_image_name = "ibm-centos-8-3-minimal-amd64-3"
+  vsi_profile = "bx2-2x8"
+  vsi_interface = "public"
+  vsi_user_data_file = "scripts/postscript.sh"
+  image_name = "packer-${local.timestamp}"
+
+  communicator = "ssh"
+  ssh_username = "root"
+  ssh_port = 22
+  ssh_timeout = "15m"
+  timeout = "30m"
+}
+
+build {
+  sources = [
+    "source.ibmcloud-vpc.centos"
+  ]
+
+  provisioner "shell" {
+    execute_command = "{{.Vars}} bash '{{.Path}}'"
+    inline = [
+      "echo 'Hello from IBM Cloud Packer Plugin'",
+      "echo 'Hello from IBM Cloud Packer Plugin' >> /hello.txt"
+    ]
+  }
+  post-processors {
+    post-processor "ibmcloud-export" {
+      image_export_job_name = "image-export-packer-1"
+      storage_bucket_name   = "storage-bucket-1"
+      format                = "qcow2"
+    }
+    post-processor "ibmcloud-export" {
+      image_id              = "r006-47ce771b-2de5-4f63-9606-c9b7421d6888"
+      image_export_job_name = "image-export-packer-2"
+      api_key               = "${var.ibm_api_key}"
+      region                = "us-south"
+      storage_bucket_name   = "storage-bucket-2"
+      format                = "qcow2"
+    }
+    post-processor "ibmcloud-export" {
+      image_id              = "r006-47ce771b-2de5-4f63-9606-c9b7421d6888"
+      image_export_job_name = "image-export-packer-3"
+      api_key               = "${var.ibm_api_key}"
+      region                = "us-south"
+      storage_bucket_name   = "storage-bucket-2"
+      format                = "qcow2"
+    }
+  }
+}
+```
+### Understanding Packer Template Blocks
+
+#### `post-processor` Block
+The `variable` block defines variables within your Packer configuration. Input variables serve as parameters for a Packer build, allowing aspects of the build to be customized without altering the build's own source code. When you declare variables in the build of your configuration, you can set their values using CLI options and environment variables.
+
+Variable | Type |Description
+--- | --- | ---
+**post-processor export image variables** |
+| |
+api_key | string | The IBM Cloud platform API key. Required.
```

Comment:
This is optional, but required only if image-id is given right ?

Reply:
Done